src/components/Header/SearchBar/SearchBar.js

```diff
@@ -1,4 +1,19 @@
-import React from 'react';
+import React, { useContext } from 'react';
+import { useDebouncedCallback } from 'use-debounce';
+import { Context as SearchContext } from '../../../contexts/SearchContext';
 import { Container } from './styles';
 
-export const SearchBar = () => <Container placeholder="Search..." />;
+export const SearchBar = () => {
+  const { dispatch } = useContext(SearchContext);
+
+  const searchDebounced = useDebouncedCallback((value) => {
+    dispatch({ type: 'search', payload: { searchText: value } });
+  }, 400);
+
+  return (
+    <Container
+      onChange={(e) => searchDebounced.callback(e.target.value)}
```

Comment:
I guess that you use use-debounce to improve performance with the search triggered by the onChange. If you want to do it without any additional libraries I suggest you listen to the enter event or add a button to trigger search on click, that could also work and will help you avoid getting to the Youtube API usage limit.